<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="point_cloud_topic"      default="/omni_stitched_cloud"/>
  <arg name="grid_map_topic"         default="/grid_map" />
  <arg name="map_frame"              default="map" />
  <arg name="map_layer_name"         default="elevation" />
  <arg name="prefix"                 default=""/>
  <arg name="set_verbosity_to_debug" default="false"/>
    
  <node name="PointCloud2_to_GridMap_msg_node" pkg="grid_map_pcl"
      type="PointCloud2_to_GridMap_msg_node" output="screen" launch-prefix="$(arg prefix)">
    <rosparam file="$(find grid_map_pcl)/config/parameters.yaml" />
    <param name="point_cloud_topic"       type="string" value="$(arg point_cloud_topic)" />
    <param name="grid_map_topic"          type="string" value="$(arg grid_map_topic)" />
    <param name="map_frame"               type="string" value="$(arg map_frame)" />
    <param name="map_layer_name"          type="string" value="$(arg map_layer_name)" />
    <param name="set_verbosity_to_debug"  type="bool" value="$(arg set_verbosity_to_debug)" />
  </node>
</launch>